### DISCUSSION TITLE
Licence problem

### DISCUSSION THREAD

**Discussion by alexec** [gavv/httpexpect#124] [Q&A] [answered]
One of your dependencies is:

https://github.com/fasthttp-contrib/websocket/issues/3

This repository does not have a LICENCE, and is therefore (presumably) copyright. 

I think that might invalidate your MIT licence. 

I am not a lawyer obviously.

As they have been unresponsive to the request for a LICENCE, you can probably sidestep this issue and replace them with the library they forked. 

https://github.com/gorilla/websocket

**Comment by gavv** [comment]
Hi!

Actually, I though I already fixed this: #74.

Why do you think `fasthttp-contrib/websocket` is still a dependency?

I searched for it and found one occurrence:

```
_examples/go.sum
33:github.com/fasthttp-contrib/websocket v0.0.0-20160511215533-1f3b11f56072 h1:DddqAaWDpywytcG8w/qoQ5sAN8X12d3Z3koB0C3Rxsc=
34:github.com/fasthttp-contrib/websocket v0.0.0-20160511215533-1f3b11f56072/go.mod h1:duJ4Jxv5lDcvg4QuQr0oowTf7dz4/CR8NtyCooz9HL8=
```

It seems it's there because `_examples` depend on `iris/v12`, and it in turn depends on `httpexpect/v2.0.0` (older release before #74), and it depends on `fasthttp-contrib/websocket`. Funny enough :)

That's not a "real" dependency, thought. It's present only in examples, and even there it's not actually used, directly or indirectly. If you vendor httpexpect, `_examples` package and its dependencies should be automatically excluded from vendoring due to leading underscore (please report a bug if it's not true anymore).

**Comment by alexec** [comment]
```
go mod graph
...
github.com/gavv/httpexpect/v2@v2.0.3 github.com/fasthttp-contrib/websocket@v0.0.0-20160511215533-1f3b11f56072
```

**Comment by alexec** [comment]
Looks like we need an upgrade:
```
 go get -u github.com/gavv/httpexpect/v2
```

**Comment by gavv** [comment, marked as answer]
Ah, so it seems you're using an older release (v2.0.3). The dependency was removed in v2.1.0. The latest release is v2.2.0.
